src/kixi/comms/components/kinesis.clj

```diff
@@ -182,7 +185,11 @@
 
   (send-command! [{:keys [producer-in-ch]} command version user payload opts]
     (when producer-in-ch
-      (async/put! producer-in-ch [:command command version user payload opts])))
+      (async/put! producer-in-ch [:command command version user payload 
+                                  (assoc opts
+                                          :command-id
+                                          (or (:kixi.comms.command/id payload)
```

Comment:
I'm not sure about this. Having a command-id in the payload is not the same as having one on the message. The `:id` option key is used to ensure a particular command ID.